File: Python/Array&String/334. Increasing Triplet Subsequence.py
""" 334. Increasing Triplet Subsequence
Difficulty: Medium
https://leetcode.com/problems/increasing-triplet-subsequence/submissions/1383516666/
 """

class Solution(object):
    def increasingTriplet(self, nums):
        smallestNum = float("inf")
        secondSmallestNum = float("inf")
        n = len(nums)

        for i in range(n):
            num = nums[i]
            if num <= smallestNum:
                smallestNum = num
            else:
                if num <= secondSmallestNum:
                    secondSmallestNum = num
                else:
                    return True
        
        return False

File: C++/Array&String/334. Increasing Triplet Subsequence.cpp
/* 334. Increasing Triplet Subsequence
Difficulty: Medium
https://leetcode.com/problems/increasing-triplet-subsequence/submissions/1030278765/
 */

class Solution {
public:
   bool increasingTriplet(vector<int>& nums) {
    int c1 = INT_MAX, c2 = INT_MAX;
    for (int x : nums) {
        if (x <= c1) {
            c1 = x;
        } else if (x <= c2) {
            c2 = x;
        } else {
            return true;
        }
    }
    return false;
}
};